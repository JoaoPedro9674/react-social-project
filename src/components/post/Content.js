import PropTypes from 'prop-types';
import React from 'react';

const Content = props => {
    const { post } = props;
    return (
        <p className="content">
            {post.content}
        </p>
    );
};

Content.propTypes = {
    post: PropTypes.object
};

export default Content;
