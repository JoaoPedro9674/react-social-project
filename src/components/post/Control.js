import React, { PropTypes } from 'react'

export class Controls extends React.Component {

}
