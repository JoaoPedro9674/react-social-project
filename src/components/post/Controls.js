import PropTypes from 'prop-types';
import React from 'react';
import { ButtonGroup, Button } from 'react-bootstrap';

const Controls = props => {
    const { post } = props;

    function showLikeCounter() {
        return post.likes > 0;
    }

    function showCommentCounter() {
        return post.comments && post.comments.length > 0;
    }

    function applyCounterClassIf(count) {
        return count ? 'control-with-counter' : null;
    }

    return (
        <ButtonGroup>
            <Button className="control">
                <i
                    className={`fa fa-thumbs-o-up ${applyCounterClassIf(showLikeCounter())}`}
                />
                {showLikeCounter() ? post.likes : null}
            </Button>
            <Button className="control">
                <i
                    className={`fa fa-comment ${applyCounterClassIf(showCommentCounter())}`}
                />
                {showCommentCounter() ? post.comments.length : null}
            </Button>
            <Button className="control">
                <i className="fa fa-share" />
            </Button>
        </ButtonGroup>
    );
};

Controls.propTypes = {
    post: PropTypes.object.isRequired
};

export default Controls;
