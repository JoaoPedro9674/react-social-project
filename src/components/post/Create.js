import React, { PropTypes } from 'react';

class CreatePost extends React.Component {
  constructor(props) {
    super(props);

    // Set up state
    this.state = {
      content: '',
    };

    // Set up event handlers
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handlePostChange = this.handlePostChange.bind(this);
  }

  handlePostChange(event) {
    const content = event.target.value;
    this.setState({ content });
  }

  handleSubmit(event) {
    event.preventDefault();
    if (this.state.content.length <= 0) {
      return;
    }
    if (this.props.onSubmit) {
      const newPost = {
        content: this.state.content,
      };
      this.props.onSubmit(newPost);
      this.setState({
        content: '',
      });
    }
  }

  render() {
    return (
      <form className="create-post" onSubmit={this.handleSubmit}>
        <textarea value={this.state.content} onChange={this.handlePostChange} placeholder="What's on your mind?" />
        <button className="btn btn-default">Post</button>
      </form>
    );
  }
}

CreatePost.propTypes = {
  onSubmit: PropTypes.func,
};

export default CreatePost;
