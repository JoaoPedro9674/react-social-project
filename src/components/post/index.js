export * from './Avatar';
export * from './Content';
export * from './Controls';
export * from './Create';
export * from './Image';
export * from './Link';
export * from './Post';
export * from './Posts';
export * from './User';
