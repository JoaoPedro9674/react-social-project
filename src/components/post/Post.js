// @flow

import React, { PropTypes } from 'react';
import Comments from '../comment/Comments';
import Controls from './Controls';

const Post = (props: Object) => {
  const { post } = props;
  return (
    <div
      className="post"
      key={post.id}
    >
      <img
        src={post.user.profilePicture}
        width={50}
        height={50}
        alt={post.content}
      />
      <a>{post.user.firstName} {post.user.lastName}</a>
      <p>
        {post.content}
      </p>
      <Comments {...props} />
      <Controls {...props} />
    </div>
  );
};

Post.propTypes = {
  post: PropTypes.object,
  user: PropTypes.object,
};

export default Post;
