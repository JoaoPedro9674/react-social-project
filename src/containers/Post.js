import React, { PropTypes, Component } from 'react';
import { AutoAffix } from 'react-overlays';

import { fetchpost } from '../shared/http';

import { Ad } from '../components/ad/Ad';
import { Post } from '../components/post/Post';
import { Link } from '../components/router/Link';

export class SinglePost extends Component {
  static propTypes = {
    params: PropTypes.shape({
      post: PropTypes.string,
    }),
  }
  constructor(props) {
    super(props);
    this.state = {
      post: null,
    };
  }

  componentDidMount() {
    fetchpost(this.props.params.post).then(post => this.setState({ post }));
  }

  render() {
    return (
      this.state.post &&
        <div className="container-fluid">
          <div className="row">
            <div className="col-sm-3">
              <Link to="/">
                <div className="pull-right">
                  <br />
                  <i className="fa fa-arrow-left" /> Back
                </div>
              </Link>
            </div>
            <div className="col-xs-12 col-sm-6">
              <Post forceOpen post={this.state.post} />
            </div>
            <div className="col-sm-2 col-xs-12 last-xs">
              <AutoAffix viewportOffsetTop={50} container={this}>
                <Ad
                  url="https://www.manning.com/books/react-in-action" imageUrl="https://drtzvj8zd0k9x.cloudfront.net/assets/ads/react+in+action+meap+ad.png"
                />
              </AutoAffix>
            </div>
          </div>
        </div>
    );
  }
}
