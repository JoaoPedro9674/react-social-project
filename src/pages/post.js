import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { loadPost } from '../actions/posts';
import Ad from '../components/ad/Ad';
import Post from '../components/post/Post';
import Link from '../components/router/Link';
import Loader from '../components/Loader';

/**
 * Component for a single post
 * @module letters/components
 * @type {Object}
 */
class SinglePost extends Component {
    static propTypes = {
        params: PropTypes.shape({
            post: PropTypes.string
        })
    };
    componentDidMount() {
        // If we don't have a post yet, dispatch an action to load it
        if (!this.props.post) {
            this.props.actions.loadPost(this.props.router.params.postId);
        }
    }
    render() {
        return this.props.post ? (
            <div className="single-post">
                <Link to="/">
                    <div className="back">
                        <i className="fa fa-arrow-left" /> Back
                    </div>
                </Link>
                <Post post={this.props.post} />
                <Ad
                    url="https://www.manning.com/books/react-in-action"
                    imageUrl="https://s3-us-west-2.amazonaws.com/react-sh/assets/ads/react+in+action+meap+ad.png"
                />
            </div>
        ) : (
            <Loader />
        );
    }
}

export default connect(
    // mapStateToProps
    (state, ownProps) => {
        console.log(ownProps);
        return {
            // try to directly read the post from our store and only fetch all posts in
            // componentDidMount if we have to
            post: state.posts[ownProps.params.postId]
        };
    },
    dispatch => {
        return {
            actions: bindActionCreators({ loadPost }, dispatch)
        };
    }
)(SinglePost);
