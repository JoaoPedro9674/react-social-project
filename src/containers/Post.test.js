import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { Post as PostModel } from '../../db/models';
import { SinglePost } from './Post';
import { Post } from '../components/post/Post';


jest.mock('../shared/http');
describe('<SinglePost/>', function () {
  it('should render correctly', () => {
    const newPost = new PostModel();
    const component = renderer.create(<SinglePost params={{post: newPost.id}} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
