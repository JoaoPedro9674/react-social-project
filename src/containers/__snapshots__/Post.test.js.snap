exports[`<SinglePost/> should render correctly 1`] = `null`;
